### ISSUE TITLE
confused about UPF port map 2152 is not used

### ISSUE THREAD

**Issue by MrPluto0** [Gradiant/5g-images#252]
In the file, UPF maps the port 2152, but in actual, the smf and gnb is connected to <docker_container_ip>:<port>, such as '172.18.0.8:2152'. I'm confusing about it.
https://github.com/Gradiant/5g-images/blob/6380d3ed7f3cf86e56abc65548135d50e4afa5ec/docs/open5gs-and-srsran-5g/core.yaml#L82

Because I'm simulating the scene as follows:
Open5gs is deployed in Server A, such as 10.111.54.59
OAI UE is deployed in Server B, such as 10.111.140.18
OAI GNB is deployed in Server C, such as 10.111.118.16
It looks like that the connection is normal, and then ue has a PDU Session 10.45.0.2, which is binded to network Interface oai_ue1. But i cannot `ping -I oai_ue1 10.45.0.1`.
By checking traffic, I found in gnb in which `172.22.0.2` is the docker container ip of upf. I know that's why i cannot ping. But it's unreasonable. 
I think it should from 10.111.140.18:2152 to 10.111.54.59:2152 which is upf map port 2152.
```
14:57:59.505543 IP (tos 0x0, ttl 64, id 44582, offset 0, flags [DF], proto UDP (17), length 128)
    10.111.140.18.2152 > 172.22.0.2.2152: [bad udp cksum 0x4318 -> 0xc9e9!] UDP, length 100
```

**Comment by avrodriguezgrad** [comment]
Hi @MrPluto0 

Have you tried network mode: host in the UPF to use the IP 10.11.54.59? Also, can you attach the config files you are using? 

BR,
Álvaro

**Comment by MrPluto0** [comment]
Thanks for reply!

Yes, I have tried network mode host. The actual stucture is:

Open5gs-C(amf, smf...) is deployed in Server A, 10.112.54.59 by dockers
Open5gs-U(upf) is deployed in Server B, 10.112.246.99 by docker-host
OAI GNB is deployed in Server C, 10.112.140.18
OAI UE is deployed in Server D, 10.112.130..180

the docker file is:
```yaml
version: '3'
services:
  mongo:
    image: mongo
    environment:
      MONGO_INITDB_DATABASE: open5gs

  amf:
    image: gradiant/open5gs:2.7.1
    command:
      - open5gs-amfd
    volumes:
      - ./configs/:/opt/open5gs/etc/open5gs/
    restart : on-failure
    ports:
      - 38412:38412/sctp

  ausf:
    image: gradiant/open5gs:2.7.1
    command:
      - open5gs-ausfd
    restart : on-failure
    volumes:
      - ./configs/:/opt/open5gs/etc/open5gs/

  bsf:
    image: gradiant/open5gs:2.7.1
    command: open5gs-bsfd
    restart : on-failure
    volumes:
      - ./configs/:/opt/open5gs/etc/open5gs/

  nrf:
    image: gradiant/open5gs:2.7.1
    command: open5gs-nrfd
    restart : on-failure
    volumes:
      - ./configs/:/opt/open5gs/etc/open5gs/

  nssf:
    image: gradiant/open5gs:2.7.1
    command: open5gs-nssfd
    restart : on-failure
    volumes:
      - ./configs/:/opt/open5gs/etc/open5gs/

  pcf:
    image: gradiant/open5gs:2.7.1
    command:
      - open5gs-pcfd
    environment:
      DB_URI: mongodb://mongo/open5gs
    restart : on-failure
    volumes:
      - ./configs/:/opt/open5gs/etc/open5gs/

  pcrf:
    image: gradiant/open5gs:2.7.1
    command: ["open5gs-pcrfd"]
    environment:
      DB_URI: mongodb://mongo/open5gs
    restart: on-failure
    volumes:
      - ./configs/:/opt/open5gs/etc/open5gs/

  scp:
    image: gradiant/open5gs:2.7.1
    command: open5gs-scpd
    environment:
      DB_URI: mongodb://mongo/open5gs
    restart: on-failure
    volumes:
      - ./configs/:/opt/open5gs/etc/open5gs/

  smf:
    image: gradiant/open5gs:2.7.1
    command: open5gs-smfd
    restart : on-failure
    volumes:
      - ./configs/:/opt/open5gs/etc/open5gs/
    ports:
      - 8805:8805/udp
      - 2152:2152/udp

  udm:
    image: gradiant/open5gs:2.7.1
    restart: on-failure
    command:
      - open5gs-udmd
    volumes:
      - ./configs/:/opt/open5gs/etc/open5gs/

  udr:
    image: gradiant/open5gs:2.7.1
    restart: on-failure
    environment:
      DB_URI: mongodb://mongo/open5gs
    command:
      - open5gs-udrd
    volumes:
      - ./configs/:/opt/open5gs/etc/open5gs/

  # upf:
  #   image: gradiant/open5gs:2.7.1
  #   command:
  #     - open5gs-upfd
  #   cap_add:
  #     - all
  #   privileged: true
  #   restart: on-failure
  #   ports:
  #     - 2152:2152
  #   volumes:
  #     - ./configs/:/opt/open5gs/etc/open5gs/

  webui:
    image: gradiant/open5gs-webui:2.7.1
    restart: on-failure
    environment:
      DB_URI: mongodb://mongo/open5gs
      NODE_ENV: development
    ports:
      - 9999:9999

networks:
  default:
    driver: bridge
    ipam:
      driver: default
      config:
        - subnet: "172.18.0.0/24"
```

```yaml
version: '3'
services:
  upf:
    image: gradiant/open5gs:2.7.1
    command:
      - open5gs-upfd
    cap_add:
      - all
    privileged: true
    restart: on-failure
    volumes:
      - ./configs/:/opt/open5gs/etc/open5gs/
    # ports:
    #   - 8805:8805/udp
    #   - 2152:2152
    network_mode: host
```

the open5gs NF config file is:

```yaml
# smf

logger:
  file: /opt/open5gs/var/log/open5gs/smf.log
#  level: info   # fatal|error|warn|info(default)|debug|trace

global:
# max:
#   ue: 1024 # The number of UE can be increased depending on memory size.
#    peer: 64

smf:
  sbi:
    server:
    - dev: eth0
      port: 7777
    client:
      #      nrf:
      #        - uri: http://127.0.0.10:7777
      scp:
      - uri: http://scp:7777
  pfcp:
    server:
    - dev: eth0
    client:
      upf:
      # - address: upf
      - address: 10.112.246.99
  gtpc:
    server:
    - dev: eth0
  gtpu:
    server:
    - dev: eth0
  metrics:
    server:
    - dev: eth0
      port: 9090
  session:
  - subnet: 10.45.0.1/16
    gateway: 10.45.0.1
    dnn: internet
  - subnet: 2001:db8:cafe::1/48
    gateway: 2001:db8:cafe::1
  dns:
  - 8.8.8.8
  - 8.8.4.4
  - 2001:4860:4860::8888
  - 2001:4860:4860::8844
  mtu: 1400
  #  p-cscf:
  #    - 127.0.0.1
  #    - ::1
  ctf:
    enabled: auto # auto(default)|yes|no
  freeDiameter: /opt/open5gs/etc/freeDiameter/smf.conf
```
```yaml
# upf

logger:
  file: /opt/open5gs/var/log/open5gs/upf.log
#  level: info   # fatal|error|warn|info(default)|debug|trace

global:
# max:
#   ue: 1024  # The number of UE can be increased depending on memory size.
#    peer: 64

upf:
  pfcp:
    server:
    - dev: eno4
    client:
      smf:     #  UPF PFCP Client try to associate SMF PFCP Server
        - address: 10.112.54.59
  gtpu:
    server:
    - dev: eno4
  session:
  - subnet: 10.45.0.1/16
    gateway: 10.45.0.1
    dnn: internet
  - subnet: 2001:db8:cafe::1/48
    gateway: 2001:db8:cafe::1
  metrics:
    server:
    - dev: eno4
      port: 9090
```

Yesterday, i tried host network, and i can `ping -I oaitun_ue1 10.45.0.1` but cannot `ping -I oaitun_ue1 baidu.com`.
Today, the same config, i found `ping -I oaitun_ue1 10.45.0.1` isn't work too.....

by `tcpdump -nn -i any port 2152`, i found the traffic flow ue -> gnb -> upf
the output in upf is:
```bash
```

**Comment by avrodriguezgrad** [comment]
Hi @MrPluto0 

Between server B - C - D are you able to do pings? Have you enabled ip_forward in the server B? It's difficult thinking which is your error, but it seems that is an error about networking and not open5gs/oai.  

On the other hand, talking about the resolution of baidu.com, it could be possible that /etc/resolv.conf is not well configured. If you got this working again, I would suggest you to check that file.

BR,
Álvaro

**Comment by MrPluto0** [comment]
Hi @avrodriguezgrad Thank you.

After trying many times, it finally works in seperate network. My configs is all right. Here is my resolution:

First, i tried open5gs core in Server A and upf use host network mode, it works. Then I found upf in Server B don't work. So I tried Server E to compare and verify. Finally, it shows server B don't work properly for open5gs.

The server B is centos. I thought docker can dismiss the difference of centos and ubuntu. But maybe the host mode of docker won't.
